src/jit/codegenlegacy.cpp

```diff
@@ -5426,6 +5426,9 @@ void CodeGen::genCodeForTreeLeaf_GT_JMP(GenTreePtr tree)
             offset += REGSIZE_BYTES;
         }
     }
+
+    // Lock the argument registers s.t. they are not trashed by code that runs before the jmp (e.g. GS cookie checks).
+    regSet.rsLockReg(fixedArgsMask);
```

Comment:
I assume the problem here is that register stress causes rsGrabReg to return a register that is used, namely an outgoing argument register set up by the GT_JMP, when doing the GS cookie check. I think the code to preserve this should be done in CodeGen::genEmitGSCookieCheck(). In fact, it already has a "preserve registers" argument, poorly named "pushReg", and an ARM TODO that is probably related to the bug you're seeing:
```
*   TODO-ARM-Bug?: pushReg is not implemented (is it needed for ARM?)
 */
void CodeGen::genEmitGSCookieCheck(bool pushReg)
```
Here, maybe you can have:
```
if (pushReg)
{
    rsLockReg...
}
```